server/src/utils/general-controller.ts

```diff
@@ -10,30 +10,53 @@ class GeneralController {
         this.name = name;
     }
 
+    /**
+     * Creates new item in model
+     * @param params {object}
+     * @param params.data {object} data to store
+     * @returns {Promise<object>} created item
+     */
     async create(params: any) {
-        const newEvent = new this.model(params.data);
-        if (await this.model.create(newEvent)) {
-            console.log(allowedFields[this.name]);
-            const event = await this.model.findById(newEvent._id, allowedFields[this.name]);
-            return event;
+        const newItem = new this.model(params.data);
```

Comment:
Why model is not typed (Model<EventModel>)? With this it would be more visible what type is coming from where.

Reply:
it can be any model not only event model